gnovm/pkg/gnolang/gotypecheck.go

```diff
@@ -142,13 +142,13 @@ const (
 // Args:
 
 // - tcmode: TypeCheckMode, see comments above.
-func TypeCheckMemPackage(mpkg *std.MemPackage, getter MemPackageGetter, pmode ParseMode, tcmode TypeCheckMode) (
+func TypeCheckMemPackage(mpkg *std.MemPackage, getter MemPackageGetter, pmode ParseMode, tcmode TypeCheckMode, height int64) (
```

Comment:
i think we shouln't add a height field, but declare a new GenesisTypeCheckMode.

Reply:
05ceb53

I added two mode, to keep information about the latest strict or relaxed and add helpers functions to handle it.
I could have use bit flags, but i think its less readable and more error prone than this explicit and easy version.